src/daemon/core.h

```diff
@@ -66,7 +67,8 @@ class t_core final
 #else
     const cryptonote::GetCheckpointsCallback& get_checkpoints = nullptr;
 #endif
-    if (!m_core.init(m_vm_HACK, nullptr, get_checkpoints))
+    const bool allow_dns = command_line::is_arg_defaulted(vm, daemon_args::arg_proxy) || command_line::get_arg(vm, daemon_args::arg_proxy_allow_dns_leaks);
```

Comment:
Should it be an error if someone sets the `proxy_allow_dns_leaks = false` flag without setting the other flag? It works as-is, but the error reporting helps the user without pointless configurations.

Reply:
Added a warning message above (I assume only showing a message is sufficient, since the behavior is unaffected). Let me know if there is a better place to put this, though.